src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/CreditManager.cs

```diff
@@ -106,7 +118,8 @@ public void Dispose()
                 {
                     while (_waiters.TryDequeue(out Waiter waiter))
                     {
-                        waiter.TaskCompletionSource.SetException(new ObjectDisposedException(nameof(CreditManager)));
+                        waiter.TaskCompletionSource.TrySetException(new ObjectDisposedException(nameof(CreditManager)));
+                        waiter.TokenRegistration.Dispose();
```

Comment:
Swap these lines to dispose of the registration first?

Reply:
With the change to `TaskCompletionSourceWithCancellation`, I've been able to remove this line altogether. So this should be resolved in the next commit.